app/components/blacklight/search_context/server_item_pagination_component.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+module Blacklight
+  module SearchContext
+    class ServerItemPaginationComponent < Blacklight::Component
+      with_collection_parameter :search_context
+
+      def initialize(search_context:, search_session:, current_document:)
+        @search_context = search_context
+        @search_session = search_session
+        @current_document_id = current_document.id
+      end
+
+      def render?
+        @search_context.present? && (@search_context[:prev] || @search_context[:next]) && (@search_session['document_id'] == @current_document_id)
+      end
+
+      ##
+      # Displays "showing X of Y items" message.
+      #
+      # @return [String]
+      def item_page_entry_info
+        t('blacklight.search.entry_pagination_info.other', current: number_with_delimiter(count),
+                                                           total: number_with_delimiter(total),
+                                                           count: total).html_safe
+      end
+
+      def link_to_previous_document(previous_document = nil, classes: 'previous', **link_opts)
+        previous_document ||= @search_context[:prev]
+        link_opts = session_tracking_params(previous_document, count - 1, per_page: per_page, search_id: search_id).merge(class: classes, rel: 'prev').merge(link_opts)
+        link_to_unless previous_document.nil?, raw(t('views.pagination.previous')), url_for_document(previous_document), link_opts do
+          tag.span raw(t('views.pagination.previous')), class: 'previous'
+        end
+      end
+
+      def link_to_next_document(next_document = nil, classes: 'next', **link_opts)
+        next_document ||= @search_context[:next]
+        link_opts = session_tracking_params(next_document, count + 1, per_page: per_page, search_id: search_id).merge(class: classes, rel: 'next').merge(link_opts)
+        link_to_unless next_document.nil?, raw(t('views.pagination.next')), url_for_document(next_document), link_opts do
+          tag.span raw(t('views.pagination.next')), class: 'next'
+        end
+      end
+
+      private
+
+      def count
+        @search_session['counter'].to_i
+      end
+
+      def per_page
+        @search_session['per_page']
+      end
+
+      def total
+        @search_session['total'].to_i
+      end
+
+      def search_id
+        controller.current_search_session&.id
```

Comment:
Yeah, I agree this seems like a mistake.  The only scenario I can think of where it might make sense is where `@search_session` is `nil` or if `@search_session` would ever differ from `controller.current_search_sesssion`.  But neither of those two scenarios makes sense to me either.

Reply:
It's unchanged from https://github.com/projectblacklight/blacklight/blob/main/app/components/blacklight/search_context_component.rb#L56-L58, but I think y'all are right that it should be changed.

Reply:
Ok, this is a little complicated: `@search_session` is a Hash representation of a search. `controller.current_search_session` is an instance of the `Search` model. I *think* we can rely on `@search_session['id']` for this value, which should be loaded in a before_action added to controllers when they include `Blacklight::Catalog`. But it relies on more assumptions than this implementation.

Again, this *was not* introduced in this PR. This is the existing behavior in `main` of a component that is renamed in this PR.

Reply:
I think it might be good to separately ticket this change.

Reply:
@barmintor I agree with you about a separate ticket vs changing it in this PR.  The implications of an instance variables vs a hash representation may be very significant in a way that is not yet clear. I keep wondering if there is a context in which these two different representation are different and it does matter which one is used at this point.  That would be bad design but still an open question.